app/addons/documents/doc-editor/components.react.jsx

```diff
@@ -218,44 +218,137 @@ var AttachmentsPanelButton = React.createClass({
+    Stores.attachmentFilterStore.on('change', this.onChange);
+  },
+
+  componentWillUnmount: function () {
+    $(window).off('resize.attachment-filter', this.updateMaxHeight);
+    Stores.attachmentFilterStore.off('change', this.onChange);
+  },
+
+  onChange: function () {
+    this.setState({
+      maxHeight: this.state.maxHeight,
+      filter: Stores.attachmentFilterStore.filter()
+    });
+  },
+
+  getMaxHeight: function () {
+    var $btn = $(this._button);
+    return $(document).height() - $btn.offset().top - $btn.outerHeight();
+  },
+
+  updateMaxHeight: function (button) {
+    if (!this._button && button) {
+      this._button = button;
+    }
+
+    this.setState({
+      maxHeight: this.getMaxHeight(),
+      filter: this.state.filter
+    });
+  },
+
   getAttachmentList: function () {
     var db = this.props.doc.database.get('id');
     var doc = this.props.doc.get('_id');
+    var attachments = this.props.doc.get('_attachments');
 
-    return _.map(this.props.doc.get('_attachments'), function (item, filename) {
+    function create(item, filename) {
       var url = FauxtonAPI.urls('document', 'attachment', db, doc, app.utils.safeURLName(filename));
       return (
-        <li key={filename}>
-          <a href={url} target="_blank" data-bypass="true"> <strong>{filename}</strong>
-            <span className="attachment-delimiter">-</span>
-            <span>{item.content_type}, {Helpers.formatSize(item.length)}</span>
-          </a>
-        </li>
+          <li key={filename} className="attachment-row">
+            <a href={url} target="_blank" data-bypass="true"> <strong>{filename}</strong>
+              <span className="attachment-delimiter">-</span>
+              <span>{item.content_type}, {Helpers.formatSize(item.length)}</span>
+            </a>
+          </li>
       );
-    });
+    }
+
+    function filter(predicate) {
+      return _.reduce(attachments, function(list, item, filename) {
+        if (predicate(filename)) {
+          list.push(create(item, filename));
+        }
+
+        return list;
+      }, []);
+    }
+
+    if (this.state.filter === '') {
+      return _.map(attachments, create);
+    } else {
+      var strong = new RegExp('^' + this.state.filter);
+      var weak = new RegExp(this.state.filter);
+      return filter(f => strong.test(f)).concat(filter(f => !strong.test(f) && weak.test(f)));
+    }
   },
 
   render: function () {
     if (this.props.isLoading || !this.props.doc.get('_attachments')) {
       return false;
     }
 
+    var ulStyle = {
+      maxHeight: this.state ? this.state.maxHeight : "none"
+    };
+
     return (
-      <div className="panel-section view-attachments-section btn-group">
+      <div className="panel-section view-attachments-section btn-group" ref={this.updateMaxHeight}>
         <button className="panel-button dropdown-toggle btn" data-bypass="true" data-toggle="dropdown" title="View Attachments"
```

Comment:
Using `ref` like this isn't actually correct. I'm not actually sure why you need to change the height?